Makefile

```diff
@@ -666,18 +666,21 @@ pylint:
 	python3 -m pylint --rcfile=$(ROOTDIR)/ci/other/pylintrc --ignore-patterns=".*\.so$$,.*\.dll$$,.*\.dylib$$" python/mxnet
 
 # MXNet extension dynamically loading libraries
-EXT_LIBS = custom_op_lib subgraph_lib
+EXT_LIBS = build/libcustomop_lib.so build/libsubgraph_lib.so
 ifeq ($(USE_CUDA), 1)
-	EXT_LIBS += custom_op_gpu_lib
+        EXT_LIBS += build/libcustomop_gpu_lib.so
 endif
 extension_libs: $(EXT_LIBS)
 
-custom_op_lib:
-	$(CXX) -shared -fPIC -std=c++11 example/extensions/lib_custom_op/gemm_lib.cc -o build/libcustomop_lib.so -I include/mxnet
-custom_op_gpu_lib:
-	$(NVCC) -shared -std=c++11 -Xcompiler -fPIC example/extensions/lib_custom_op/relu_lib.cu -o build/libcustomop_gpu_lib.so -I include/mxnet
-subgraph_lib:
-	$(CXX) -shared -fPIC -std=c++11 example/extensions/lib_subgraph/subgraph_lib.cc -o build/libsubgraph_lib.so -I include/mxnet
+build/libcustomop_lib.so:
```

Comment:
Met the same problem on my local machine. But curious to know why it passed the CI?

Reply:
Which problem did you get on your local machine? can you share the error message/log?

Reply:
With a clean master branch:
```bash
[lvtao@mlt2-clx092 mxnet-temp]$ make custom_op_lib
Makefile:229: "USE_LAPACK disabled because libraries were not found"
Makefile:357: WARNING: Significant performance increases can be achieved by installing and enabling gperftools or jemalloc development packages
g++ -shared -fPIC -std=c++11 example/extensions/lib_custom_op/gemm_lib.cc -o build/libcustomop_lib.so -I include/mxnet
/usr/bin/ld: cannot open output file build/libcustomop_lib.so: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [custom_op_lib] Error 1
```

I think this problem can be fixed by this PR, but I'm wondering why it's not caught by CI.

Reply:
it may be caused by different make build flag, so that `build` folder is created before the building of custom_op targets. after the fix it should run with any build flags

Reply:
I can see this error even not pass any build target to make. That means to call `make -j4` directly.

Reply:
if you don't pass any flag, make will read the config in make/config.mk as default. CI overwrote a handful of flags when running so it may block the error